Extensions/Multiplayer/messageManager.ts

```diff
@@ -1768,7 +1862,17 @@ namespace gdjs {
         messages.forEach((message) => {
           const messageData = message.getData();
           const messageSender = message.getSender();
-          runtimeScene.getGame().updateFromNetworkSyncData(messageData);
+          if (gdjs.multiplayer.isReadyToSendOrReceiveGameUpdateMessages()) {
+            runtimeScene.getGame().updateFromNetworkSyncData(messageData);
+          } else {
+            // If the game is not ready to receive game update messages, we need to save the data for later use.
+            // This can happen when joining a game that is already running.
+            lastReceivedGameSyncDataUpdates.push(messageData);
+            if (lastReceivedGameSyncDataUpdates.length > 10) {
+              // We only keep the last 10 updates.
+              lastReceivedGameSyncDataUpdates.shift();
+            }
```

Comment:
This kind of code would almost mandate to be in a dedicated data structure so it's clearer for the reader that this is just a "either we update from what we received, or we store it for later":

```suggestion
            lastReceivedGameSyncDataUpdates.store(messageData);
```

Reply:
done